rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -705,10 +733,20 @@ void SqliteStorage::read_metadata()
         rcutils_time_point_value_t, std::string>();
 
       for (auto result : query_results) {
+        std::vector<rclcpp::QoS> offered_qos_profiles {};
```

Comment:
Here, a conversion from serialized YAML (`std::string`) & `std::vector<rclcpp::QoS>` is needed.